cpp/src/io/json/json_column.cu

```diff
@@ -525,14 +525,15 @@ void make_device_json_column(device_span<SymbolT const> input,
       auto parent_node_id = ordered_parent_node_ids[i];
       if (parent_node_id != parent_node_sentinel and node_categories[parent_node_id] == NC_LIST) {
         // unique item
-        if (i == 0 ||
+        if (i == 0 or
             (col_ids[i - 1] != col_ids[i] or ordered_parent_node_ids[i - 1] != parent_node_id)) {
           // scatter to list_offset
           d_columns_data[original_col_ids[parent_node_id]]
             .child_offsets[row_offsets[parent_node_id]] = ordered_row_offsets[i];
         }
         // TODO: verify if this code is right. check with more test cases.
```

Comment:
Is this TODO still needed? Or does this PR cover enough test cases that we now trust its correctness?

Reply:
will keep the TODO for now. 
This PR has test cases to test the conditions. But it needs a few more test cases to be thorough.